src/Promise.php

```diff
@@ -10,14 +10,9 @@
     /**
      * Registers a callback to be invoked when the promise is resolved.
      *
-     * The callback receives `null` as first parameter and `$value` as second parameter on success. It receives the
-     * failure reason as first parameter and `null` as second parameter on failure.
-     *
      * If the promise is already resolved, the callback MUST be executed immediately.
      *
-     * Warning: If you use type declarations for `$value`, be sure to make them accept `null` in case of failures.
```

Comment:
I think this warning should stay in place, it has too often been an error to miss the nullability.